streams/examples/main/java/org/apache/kafka/streams/examples/WordCount/WordCountProcessorJob.java

```diff
@@ -49,17 +48,21 @@
                 public void init(ProcessorContext context) {
                     this.context = context;
                     this.context.schedule(1000);
-                    this.kvStore = (KeyValueStore<String, Integer>) context.getStateStore("LOCAL-STATE");
+                    this.kvStore = (KeyValueStore<String, Integer>) context.getStateStore("Counts");
                 }
 
                 @Override
-                public void process(String key, String value) {
-                    Integer oldValue = this.kvStore.get(key);
-                    Integer newValue = Integer.parseInt(value);
-                    if (oldValue == null) {
-                        this.kvStore.put(key, newValue);
-                    } else {
-                        this.kvStore.put(key, oldValue + newValue);
+                public void process(String dummy, String line) {
+                    String words[] = line.toLowerCase().split(" ");
+
+                    for (String word : words) {
+                        Integer oldValue = this.kvStore.get(word);
+
+                        if (oldValue == null) {
```

Comment:
Yup we can do that (you mean `putIfAbsent` right?). Since a store will only be accessed by one thread there should not be any atomicity issues. Filed https://issues.apache.org/jira/browse/KAFKA-3133